fast64_internal/f3d/f3d_bleed.py

```diff
@@ -40,9 +45,73 @@ class BleedGraphics:
     # bleed_state "enums"
     bleed_start = 1
     bleed_in_progress = 2
+    # tells bleed logic to check against itself instead of last list
+    bleed_self_conflict = 3
 
     def __init__(self):
         self.bled_gfx_lists = dict()
+        # build world default cmds to compare against, f3d types needed for reset cmd building
+        self.is_f3d_old = bpy.context.scene.f3d_type == "F3D"
+        self.is_f3dex2 = "F3DEX2" in bpy.context.scene.f3d_type
+        self.build_default_geo()
+        self.build_default_othermodes()
+
+    def build_default_geo(self):
+        defaults = bpy.context.scene.world.rdp_defaults
+
+        setGeo = SPSetGeometryMode([])
+        clearGeo = SPClearGeometryMode([])
+
+        def place_in_flaglist(flag: bool, enum: str, set_list: SPSetGeometryMode, clear_list: SPClearGeometryMode):
+            if flag:
+                set_list.flagList.append(enum)
+            else:
+                clear_list.flagList.append(enum)
+
+        place_in_flaglist(defaults.g_zbuffer, "G_ZBUFFER", setGeo, clearGeo)
+        place_in_flaglist(defaults.g_shade, "G_SHADE", setGeo, clearGeo)
+        place_in_flaglist(defaults.g_cull_front, "G_CULL_FRONT", setGeo, clearGeo)
+        place_in_flaglist(defaults.g_cull_back, "G_CULL_BACK", setGeo, clearGeo)
+        place_in_flaglist(defaults.g_fog, "G_FOG", setGeo, clearGeo)
+        place_in_flaglist(defaults.g_lighting, "G_LIGHTING", setGeo, clearGeo)
+        place_in_flaglist(defaults.g_tex_gen, "G_TEXTURE_GEN", setGeo, clearGeo)
+        place_in_flaglist(defaults.g_tex_gen_linear, "G_TEXTURE_GEN_LINEAR", setGeo, clearGeo)
+        place_in_flaglist(defaults.g_shade_smooth, "G_SHADING_SMOOTH", setGeo, clearGeo)
+        if bpy.context.scene.f3d_type == "F3DEX_GBI_2" or bpy.context.scene.f3d_type == "F3DEX_GBI":
+            place_in_flaglist(defaults.g_clipping, "G_CLIPPING", setGeo, clearGeo)
+
+        self.default_load_geo = SPLoadGeometryMode(setGeo.flagList)
+        self.default_set_geo = setGeo
+        self.default_clear_geo = clearGeo
+
+    def build_default_othermodes(self):
+        is_hardware_v1 = bpy.context.scene.isHWv1
+        defaults = bpy.context.scene.world.rdp_defaults
+
+        othermode_H = SPSetOtherMode("G_SETOTHERMODE_H", 4, 20 - self.is_f3d_old, [])
```

Comment:
I'm aware of this. The way I wrote it is how it is done for all other cases of `SPSetOtherMode` and I didn't bother to change it. I guess I could change this but it seemed unrelated to the PR overall.

Reply:
it's fine tbh, I said that just in case you didn't know about the field initialiser but if you know about it then I'm cool with it
I guess that would be part of a F3D cleanup